chain-abci/src/slashing.rs

```diff
@@ -11,23 +12,31 @@ pub struct SlashingSchedule {
     pub slash_ratio: SlashRatio,
     /// Time after which slashing can be performed
     pub slashing_time: Timespec,
+    /// Kind of punishment for which the account is scheduled to be slashed
+    pub punishment_kind: PunishmentKind,
 }
 
 impl SlashingSchedule {
     /// Creates a new instance of `SlashingSchedule`
     #[inline]
-    pub fn new(slash_ratio: SlashRatio, slashing_time: Timespec) -> Self {
+    pub fn new(
+        slash_ratio: SlashRatio,
+        slashing_time: Timespec,
+        punishment_kind: PunishmentKind,
+    ) -> Self {
         Self {
             slash_ratio,
             slashing_time,
+            punishment_kind,
         }
     }
 
     /// Updates slash ratio (only if proposed ratio is greater than current)
     #[inline]
-    pub fn update_slash_ratio(&mut self, new_ratio: SlashRatio) {
+    pub fn update_slash_ratio(&mut self, new_ratio: SlashRatio, punishment_kind: PunishmentKind) {
         if new_ratio > self.slash_ratio {
-            self.slash_ratio = new_ratio
+            self.slash_ratio = new_ratio;
+            self.punishment_kind = punishment_kind;
```

Comment:
seems this isn't tested -- perhaps add a test where two infractions happen?

Reply:
Added a test for checking multiple validator faults